homework/tests/example.spec.js

```diff
@@ -12,20 +12,424 @@ const data = [
   '7',
   '8',
   '9'
+
 ]
 
+const operations = [
+  'Add',
+  'Subtract',
+  'Multiply',
+  'Divide',
+  'Concatenate'
+]
 
 data.forEach(version => {
-  test.describe(version + ': Add', () => {
-    test('Concatenating 2 and 3 results in 23', async ({ page }) => {
-      await page.goto('https://testsheepnz.github.io/BasicCalculator');
-      await page.selectOption('#selectBuild', { label: version});
-      await page.locator('#number1Field').type('2');
-      await page.locator('#number2Field').type('3');
-      await page.selectOption('#selectOperationDropdown', {label: 'Add'});
-      await page.locator('#calculateButton').click();
-  
-      await expect(page.locator('#numberAnswerField')).toHaveValue('5');
+
+  test.describe('Calculator build ' + version, () => {
+
+    test.describe('Calculator elements ', () => {
+      test('Should be able to see the "First Number" field', async ({ page }) => {
+            let searchCalculator= new SearchCalculator(page);
+            await searchCalculator.navigate();
+            await searchCalculator.selectVersion(version);
+
+            let searchResultsCalculator = new SearchResultsCalculator(page);
+            let state = await searchResultsCalculator.getButtonState('#number1Field');
+
+            await expect(state, 'The first number field is not shown.' ).toBe(false);
+          });
+
+          test('Should be able to see the "Second number" field', async ({ page }) => {
+            let searchCalculator= new SearchCalculator(page);
+            await searchCalculator.navigate();
+            await searchCalculator.selectVersion(version);
+
+            let searchResultsCalculator = new SearchResultsCalculator(page);
+            let state = await searchResultsCalculator.getButtonState('#number2Field');
+
+            await expect(state, 'The second number field is not shown.' ).toBe(false);
+          });
+
+          test('Should respect the order of "First number" and "Second number" in subtraction', async ({ page }) => {
+            let searchCalculator= new SearchCalculator(page);
+            await searchCalculator.navigate();
+            await searchCalculator.selectVersion(version);
+
+            let searchResultsCalculator = new SearchResultsCalculator(page);
+            let answer = await searchResultsCalculator.getOperationResults('30','20','Subtract')
+
+            await expect(answer, 'The order of the First and Second number field in calculation is not correct.' ).not.toHaveValue('-10');
```

Comment:
taip yra labai didelė tikimybė praleisti kritines klaidas į produkciją, perdaryk šį patikrinimą (kitose vietose kur naudoji not.toHaveValue irgi perdaryk)